addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -98,9 +125,10 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
           this.env.pos.config.gift_card_product_id[0]
         ];
 
+      let currentGiftCardId = await this.getGiftCard().id;
```

Comment:
I think this should be `let currentGiftCardId = (await this.getGiftCard()).id;`.
Without the extra parentheses, when trying to use again the same gift card, the error message is not shown and a gift card line is added.

But I think it should be like this to be more safe:
```js
let currentGiftCard = await this.getGiftCard();
let currentGiftCardId = currentGiftCard && currentGiftCard.id;
```